oximeter/db/src/client/oxql.rs

```diff
@@ -198,6 +199,59 @@ impl Client {
         result
     }
 
+    /// Run a OxQL query.
+    pub async fn oxql_query_project(
+        &self,
+        query: impl AsRef<str>,
+        silo_id: Uuid,
+        project_id: Uuid,
+    ) -> Result<OxqlResult, Error> {
+        let query = query.as_ref();
+
+        let parsed_query = oxql::Query::new(query)?;
+        // this has to be used for everything
+        let filtered_query = parsed_query.add_filters(vec![
+            uuid_eq_filter("silo_id".to_string(), silo_id),
+            uuid_eq_filter("project_id".to_string(), project_id),
+        ]);
+
+        let plan = self.build_query_plan(&filtered_query).await?;
+        if plan.requires_full_table_scan() {
+            return Err(Error::Oxql(anyhow::anyhow!(
+                "This query requires at least one full table scan. \
+                Please rewrite the query to filter either the fields \
+                or timestamps, in order to reduce the amount of data \
+                fetched from the database."
+            )));
+        }
+        let query_id = Uuid::new_v4();
+        let query_log =
+            self.log.new(slog::o!("query_id" => query_id.to_string()));
+        debug!(
+            query_log,
+            "parsed OxQL query";
+            "query" => query,
+            "filtered_query" => ?filtered_query,
+            "parsed_query" => ?parsed_query,
+        );
+        let id = usdt::UniqueId::new();
+        probes::oxql__query__start!(|| (&id, &query_id, query));
+        let mut total_rows_fetched = 0;
+        let result = self
+            .run_oxql_query(
+                &query_log,
+                &mut self.claim_connection().await?,
+                query_id,
+                filtered_query,
+                &mut total_rows_fetched,
+                None,
+                None,
+            )
+            .await;
+        probes::oxql__query__done!(|| (&id, &query_id));
+        result
+    }
+
```

Comment:
This duplicates almost the entire function above (`oxql_query`) -- I should probably extract the shared code.

Reply:
I think that's a good idea. One way might be to have the main method accept an enum like this:

```rust
enum QueryAuthzScope {
    Fleet,
    Silo { silo_id: Uuid },
    Project { silo_id: Uuid, project_id: Uuid },
}
```

`Fleet` would mean the existing `oxql_query()` method on `main`, which does no scoping or adding of filters. The other two would add in the specific filters as needed, like you're doing here. They'd all then call the same actual method to run the query, which is basically the one we have today that does not modify the query at all.